core/src/main/java/org/elasticsearch/script/ScriptService.java

```diff
@@ -646,7 +645,17 @@ public ParseField getParseField() {
             return parseField;
         }
 
-        public ScriptMode getDefaultScriptMode() {
+        public ScriptMode getDefaultScriptMode(@Nullable String language) {
+            if (language != null) {
```

Comment:
I wonder if it makes more sense to push the default into the engine rather than a fixed switch statement. Like, plugins could declare that they should be on by default or off by default.